php-classes/Slate/Courses/Section.php

```diff
@@ -411,6 +419,68 @@ public function getCohorts()
         }
     }
 
+    public function findBlogTags()
+    {
+        return TagItem::getTagsSummary([
+            'Class' => BlogPost::class,
+            'classConditions' => [
+                'ContextClass' => static::getStaticRootClass(),
+                'ContextID' => $this->ID
+            ]
+        ]);
+    }
+
+    public function findBlogPosts($conditions, $limit, $offset, $tag)
+    {
+      $options = [
+        'limit' => $limit,
+        'offset' => $offset,
+        'calcFoundRows' => 'yes',
+        'conditions' => $conditions
+      ];
+
+      if ($tag!=null) {
+
+          $tagIDsQuery = 'SELECT ContextID FROM `tag_items` WHERE (`ContextClass` = "%s") AND (`TagID` = %u)';
+          $tagIDsConditions = [
+              'ContextClass' => DB::escape(BlogPost::getStaticRootClass()),
+              'TagID' => $tag->ID
+          ];
+
+          $tagItemIDs = DB::allValues('ContextID', $tagIDsQuery, $tagIDsConditions);
```

Comment:
`$conditions` and the `'Field' => 'value'` syntax is commonly used when passing a conditions structure to high-level ActiveRecord methods that parse that structure to generate SQL -- most importantly escaping is handled automatically in these cases

In this case, low-level `DB` methods don't have any of those smarts, they are just wrapping around `sprintf`. The use of the name `conditions` is misleading, and using named keys is misleading because they are ignored and this only works because of the position of the two args

I would prefer this form:

```suggestion
          $tagIDsQuery = ;
          $tagIDsConditions = [
          ];

          $tagItemIDs = DB::allValues(
              'ContextID',
              'SELECT ContextID FROM `tag_items` WHERE (`ContextClass` = "%s") AND (`TagID` = %u)',
              [
                  DB::escape(BlogPost::getStaticRootClass(),
                  $tag->ID
              ]
          );
```

As it makes the positional connection clearer